xcdat/dataset.py

```diff
@@ -444,25 +401,91 @@ def decode_non_cf_time(dataset: xr.Dataset) -> xr.Dataset:
+
+
+def _postprocess_dataset(
+    dataset: xr.Dataset,
+    data_var: Optional[str] = None,
+    center_times: bool = False,
+    add_bounds: bool = False,
+    lon_orient: Optional[Tuple[float, float]] = None,
+) -> xr.Dataset:
+    """Post-processes a Dataset object.
+
+    Parameters
+    ----------
+    dataset : xr.Dataset
+        The dataset.
+    data_var: Optional[str], optional
+        The key of the data variable to keep in the Dataset, by default None.
+    center_times: bool, optional
+        If True, center time coordinates using the midpoint between its upper
+        and lower bounds. Otherwise, use the provided time coordinates, by
+        default False.
+    add_bounds: bool, optional
+        If True, add bounds for supported axes (T, X, Y) if they are missing in
+        the Dataset, by default False.
+    lon_orient: Optional[Tuple[float, float]], optional
+        The orientation to use for the Dataset's longitude axis (if it exists),
+        by default None.
+
+        Supported options:
+
+          * None:  use the current orientation (if the longitude axis exists)
+          * (-180, 180): represents [-180, 180) in math notation
+          * (0, 360): represents [0, 360) in math notation
+
+    Returns
+    -------
+    xr.Dataset
+        The Dataset.
+
+    Raises
+    ------
+    ValueError
+        If ``center_times==True`` but there are no time coordinates.
+    ValueError
+        If ``lon_orient is not None`` but there are no longitude coordinates.
+    """
+    if data_var is not None:
+        dataset = _keep_single_var(dataset, data_var)
+
+    if center_times:
+        if dataset.cf.dims.get("T") is not None:
+            dataset = dataset.temporal.center_times(dataset)
+        else:
+            raise ValueError("This dataset does not have a time coordinates to center.")
+
+    if add_bounds:
+        dataset = dataset.bounds.add_missing_bounds()
+
+    if lon_orient is not None:
+        if dataset.cf.dims.get("X") is not None:
+            dataset = swap_lon_axis(dataset, to=lon_orient, sort_ascending=True)
+        else:
+            raise ValueError(
+                "This dataset does not have longitude coordinates to reorient."
+            )
 
     return dataset
 
 
-def _keep_single_var(dataset: xr.Dataset, data_var: Optional[str]) -> xr.Dataset:
-    """Keep a single data variable in the Dataset.
+def _keep_single_var(dataset: xr.Dataset, key: str) -> xr.Dataset:
+    """Keeps a single non-bounds data variable in the Dataset.
 
-    If ``data_var`` is not None, then this function checks if the ``data_var``
-    exists in the Dataset and if it is a regular data var. If those checks pass,
-    it will subset the Dataset to retain that ``data_var`` and all bounds data
-    vars.
+    This function checks if the ``data_var`` key exists in the Dataset and
+    if it is non-bounds. If those checks pass, it will subset the Dataset to
+    retain that non-bounds ``data_var`` and all bounds data vars.
```

Comment:
```suggestion
    This function checks if the ``data_var`` key exists in the Dataset and
    it is not related to bounds. If those checks pass, it will subset the 
    Dataset to retain that non-bounds ``data_var`` and all bounds data vars.
```